src/omniperf_profile/profiler_base.py

```diff
@@ -27,19 +27,215 @@
 import glob
 import sys
 import os
-from utils.utils import capture_subprocess_output, run_prof, gen_sysinfo, run_rocscope, error
+import re
+from utils.utils import capture_subprocess_output, run_prof, gen_sysinfo, run_rocscope, error, demarcate
 import config
+import pandas as pd
 
 class OmniProfiler_Base():
-    def __init__(self,args, profiler_mode,soc):
+    def __init__(self, args, profiler_mode, soc):
         self.__args = args
         self.__profiler = profiler_mode
-        self.__soc = soc
-        
+        self._soc = soc # OmniSoC obj
         self.__perfmon_dir = os.path.join(str(config.omniperf_home), "omniperf_soc", "profile_configs")
 
     def get_args(self):
         return self.__args
+    def get_profiler_options(self, fname):
+        """Fetch any version specific arguments required by profiler
+        """
+        # assume no SoC specific options and return empty list by default
+        return []
+    
+    @demarcate
+    def pmc_perf_split(self):
+        """Avoid default rocprof join utility by spliting each line into a separate input file
+        """
+        workload_perfmon_dir = os.path.join(self.__args.path, "perfmon")
+        lines = open(os.path.join(workload_perfmon_dir, "pmc_perf.txt"), "r").read().splitlines()
+
+        # Iterate over each line in pmc_perf.txt
+        mpattern = r"^pmc:(.*)"
+        i = 0
+        for line in lines:
+            # Verify no comments
+            stext = line.split("#")[0].strip()
+            if not stext:
+                continue
+
+            # all pmc counters start with  "pmc:"
+            m = re.match(mpattern, stext)
+            if m is None:
+                continue
+
+            # Create separate file for each line
+            fd = open(workload_perfmon_dir + "/pmc_perf_" + str(i) + ".txt", "w")
+            fd.write(stext + "\n\n")
+            fd.write("gpu:\n")
+            fd.write("range:\n")
+            fd.write("kernel:\n")
+            fd.close()
+
+            i += 1
+
+        # Remove old pmc_perf.txt input from perfmon dir
+        os.remove(workload_perfmon_dir + "/pmc_perf.txt")
+
+    # joins disparate runs less dumbly than rocprof
```

Comment:
let's be nice.

`# joins disparate runs by hand instead of relying on rocprof`